app/models/version.rb

```diff
@@ -309,6 +309,11 @@ def recalculate_metadata!
     nil
   end
 
+  def link_uri(name, linkset = nil)
+    metadata[name.to_s].presence || linkset.try(name).presence ||
+      (name.to_s == "documentation_uri" ? documentation_path : nil)
```

Comment:
instead of this condition, can we add the `documentation_uri` method on the linkset?